sdk/core/src/main/java/com/klaviyo/coresdk/config/SystemClock.kt

```diff
@@ -0,0 +1,29 @@
+package com.klaviyo.coresdk.config
+
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.TimeZone
+import java.util.Timer
+import kotlin.concurrent.schedule
+
+internal object SystemClock : Clock {
+
+    private val format = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").apply {
+        timeZone = TimeZone.getTimeZone("UTC")
+    }
+
+    override fun currentTimeMillis(): Long {
+        return System.currentTimeMillis()
+    }
+
+    override fun currentTimeAsString(): String {
+        return format.format(Date(currentTimeMillis()))
+    }
+
+    override fun schedule(delay: Long, task: () -> Unit): Clock.Cancellable = Timer()
+        .schedule(delay) { task() }.let {
+            object : Clock.Cancellable {
+                override fun cancel() = it.cancel()
+            }
+        }
```

Comment:
since task is a lambda, you can just do `.schedule(delay, task)` no need for the extra `{ task() }`

Reply:
I might be missing something, but that doesn't seem to work. `Timer.schedule` expects a `TimerTask.()->Unit` (what??) so my lambda doesn't convert without using the closure.

Reply:
interesting, yeah just played around locally and seems the only way is to change the type:
```
fun schedule(delay: Long, task: TimerTask.() -> Unit) =
    Timer().schedule(delay=delay, task)
```
but you can still schedule with a lambda as before:
```
schedule(10L) { // this context is a TimerTask
    something()
}
```
that said, not sure if we want the interface exposing the `TimerTask` to the lambda so this is fine